atomic_reactor/inner.py

```diff
@@ -326,8 +326,11 @@ def build_docker_image(self):
 
             self.build_failed = build_result.is_failed()
 
-            if not build_result.is_failed():
-                self.built_image_inspect = self.builder.inspect_built_image()
+            if build_result.is_failed():
+                # The docker build failed
+                return build_result
```

Comment:
It will. The `finally` clause is guaranteed to run when we exit the block. I can update the comment.